app/lib/credentialVerificationStatus.ts

```diff
@@ -0,0 +1,77 @@
+import { VerifyPayload } from './verifiableObject';
+
+export type VerificationStatus = 'verifying' | 'verified' | 'warning' | 'not_verified';
+
+/**
+ * Determines the verification status of a credential based on its verification result.
+ * 
+ * @param verifyPayload - The verification payload from useVerifyCredential
+ * @returns The verification status: 'verified', 'warning', 'not_verified', or 'verifying'
+ * 
+ * Status definitions:
+ * - 'verified': All checks passed (valid signature, not revoked, not expired, registered issuer)
+ * - 'warning': Non-critical issues (expired or unregistered issuer) but signature is valid
+ * - 'not_verified': Critical failures (invalid signature or revocation status failed)
+ * - 'verifying': Verification is in progress
+ */
+export function getVerificationStatus(verifyPayload: VerifyPayload | null): VerificationStatus {
```

Comment:
This logic should be moved to https://github.com/digitalcredentials/verifier-core